src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
-                _setupFlowViewModel.StartAppManagementFlow(query);
-                await SearchAndSelectAsync(query);
+                await AppActivationFlowAsync(wingetIdsQuery, ActivationQueryType.WingetIds);
             }
         }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
-        }
     }
 
-    private async Task SearchAndSelectAsync(string query)
+    private async Task AppActivationFlowAsync(string query, ActivationQueryType queryType)
     {
-        var parameters = HttpUtility.ParseQueryString(query);
-        var searchParameter = parameters["search"];
+        if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+        {
+            _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
+            await _mainWindow.ShowErrorMessageDialogAsync(
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.AppInstallActivationTitle),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.URIActivationFailedBusy),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.Close));
+            return;
+        }
 
-        if (string.IsNullOrEmpty(searchParameter))
+        var identifiers = ParseIdentifiers(query, queryType);
+        if (identifiers.Length == 0)
         {
-            _log.Warning("Search parameter is missing or empty in the query.");
+            _log.Warning("No valid identifiers provided in the query.");
             return;
         }
 
-        // Currently using the first search term only
-        var firstSearchTerm = searchParameter.Split(Separator, StringSplitOptions.RemoveEmptyEntries)
-                                             .Select(term => term.Trim(' ', '"'))
-                                             .FirstOrDefault();
+        _log.Information("Starting add-apps-to-cart activation");
+        _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
+        _setupFlowViewModel.StartAppManagementFlow(queryType == ActivationQueryType.Search ? identifiers[0] : null);
+        await HandleAppSelectionAsync(identifiers, queryType);
+    }
 
-        if (string.IsNullOrEmpty(firstSearchTerm))
+    private async Task HandleAppSelectionAsync(string[] identifiers, ActivationQueryType queryType)
+    {
+        try
         {
-            _log.Warning("No valid search term was extracted from the query.");
-            return;
+            await SearchAndSelectAsync(identifiers);
         }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
+        }
+    }
+
+    private string[] ParseIdentifiers(string query, ActivationQueryType queryType)
+    {
+        switch (queryType)
+        {
+            case ActivationQueryType.Search:
```

Comment:
Why are these two cases treated differently?

Reply:
in the case of search, we expect a "natural language" search query, we take in only one and then pre-fill the search box in the UX. In the new case of Winget Package URIs, we're expecting exact URIs that evaluate to precise packages that we simply add to cart (as many as are specified).